project/src/test/java/com/google/sps/EventRankerTest.java

```diff
@@ -0,0 +1,159 @@
+import com.google.sps.utilities.EventRanker;
+import com.google.sps.utilities.TestUtils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Tests the ranking algorithm for event discovery */
+@RunWith(JUnit4.class)
+public final class EventRankerTest {
+  private static final String CONSERVATION = "conservation";
+  private static final String FOOD = "food";
+  private static final String MUSIC = "music";
+  private static final String SEWING = "sewing";
+  private static final Set<String> INTERESTS_CONSERVATION_FOOD =
+      new HashSet<>(Arrays.asList(CONSERVATION, FOOD));
+  private static final Set<String> SKILLS_MUSIC = new HashSet<>(Arrays.asList(MUSIC));
+  private static Event EVENT_CONSERVATION_FOOD_MUSIC;
+  private static Event EVENT_FOOD_MUSIC;
+  private static Event EVENT_CONSERVATION_MUSIC;
+  private static Event EVENT_FOOD;
+  private static Event EVENT_SEWING;
+  private static User USER_CONSERVATION_FOOD_MUSIC;
+  private static User USER_NO_INTERESTS_OR_SKILLS;
+  private static VolunteeringOpportunity OPPORTUNITY_MUSIC;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    int currentYear = new java.util.Date().getYear();
+    EVENT_CONSERVATION_FOOD_MUSIC =
+        TestUtils.newEvent().toBuilder()
+            .setLabels(new HashSet<>(Arrays.asList(CONSERVATION, FOOD, MUSIC)))
+            .build();
+    EVENT_FOOD_MUSIC =
+        TestUtils.newEvent().toBuilder()
+            .setLabels(new HashSet<>(Arrays.asList(FOOD, MUSIC)))
+            .build();
+    EVENT_CONSERVATION_MUSIC =
+        TestUtils.newEvent().toBuilder()
+            .setLabels(new HashSet<>(Arrays.asList(CONSERVATION, MUSIC)))
+            .build();
+    EVENT_FOOD =
+        TestUtils.newEvent().toBuilder().setLabels(new HashSet<>(Arrays.asList(FOOD))).build();
+    EVENT_SEWING =
+        TestUtils.newEvent().toBuilder().setLabels(new HashSet<>(Arrays.asList(SEWING))).build();
+    USER_CONSERVATION_FOOD_MUSIC =
+        TestUtils.newUser().toBuilder()
+            .setInterests(INTERESTS_CONSERVATION_FOOD)
+            .setSkills(SKILLS_MUSIC)
+            .build();
+    USER_NO_INTERESTS_OR_SKILLS =
+        TestUtils.newUser().toBuilder().setInterests(new HashSet<>()).setSkills(new HashSet<>()).build();
+    OPPORTUNITY_MUSIC =
+        TestUtils.newVolunteeringOpportunityWithEventId(EVENT_FOOD_MUSIC.getId());
+    EVENT_FOOD_MUSIC = EVENT_FOOD_MUSIC.toBuilder().addOpportunity(OPPORTUNITY_MUSIC).build();
+  }
+
+  @Test
+  public void testRankingEmptyEvents() throws IOException {
+    Assert.assertEquals(
+        new ArrayList<Event>(),
+        EventRanker.rankEvents(USER_CONSERVATION_FOOD_MUSIC, new HashSet<Event>()));
```

Comment:
Try to reduce the information noise in your test. For example in this test, you just need to use a random User, not a User with food interests and music skill.

So I would use USER or newUser() here.

Reply:
That makes sense, done